jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -489,7 +489,9 @@ private void checkClassAccess(Class<?> targetClass) throws IllegalAccessExceptio
 					return;
 				}
 			} else {
-				if (((PACKAGE == (accessMode & PACKAGE)) || Modifier.isPrivate(accessMode)) && isSamePackage(accessClass, targetClass)) {
+				if (((PACKAGE == (accessMode & PACKAGE)) || Modifier.isPrivate(accessMode))
```

Comment:
I'm having a hard time figuring out if this change is correct or not because the original function seems confused about its purpose.

It's called `checkClassAccess` but is checking for `isProtected` on a class's modifiers.... Classes can only be `public` or `package`.  Nothing should ever end up in the `protected` case as it should be dead code.

The use of access mode to determine if a `package` class is accessible also is confusing.

Overall, this needs more thought.